test/integration/application.test.ts

```diff
@@ -0,0 +1,403 @@
+const userData = require("../fixtures/user/user")();
+const applicationModel = require("../../models/applications");
+const { requestRoleData } = require("../fixtures/discordactions/discordactions");
+
+const applicationsData = require("../fixtures/applications/applications")();
+const cookieName = config.get("userToken.cookieName");
+
+const appOwner = userData[3];
+const superUser = userData[4];
+const secondUser = userData[0];
+
+chai.use(chaiHttp);
+
+let userId: string;
+let superUserId: string;
+let secondUserId: string;
+let jwt: string;
+let superUserJwt: string;
+let secondUserJwt: string;
+let applicationId1: string;
+let applicationId2: string;
+let applicationId3: string;
+let applicationId4: string;
+let applicationId5: string;
+
+describe("Application", function () {
+  before(async function () {
+    const userIdPromises = [addUser(appOwner), addUser(superUser), addUser(secondUser)];
+    const [userId1, userId2, userId3] = await Promise.all(userIdPromises);
+    userId = userId1;
+    superUserId = userId2;
+    secondUserId = userId3;
+    jwt = authService.generateAuthToken({ userId });
+    superUserJwt = authService.generateAuthToken({ userId: superUserId });
+    secondUserJwt = authService.generateAuthToken({ userId: secondUserId });
+    const applicationOne = { ...applicationsData[0], userId };
+    const applicationTwo = { ...applicationsData[1], userId: superUserId };
+    const applicationThree = { ...applicationsData[2], userId: "fakfjdkfjkfasjdkfsjdkf" };
+    const applicationFour = { ...applicationsData[3], userId: "fkasdjfkldjfldjkfalsdfjl" };
+    const applicationFive = { ...applicationsData[4], userId: "kfasdjfkdlfjkasdjflsdjfk" };
+
+    const promises = [
+      applicationModel.addApplication(applicationOne),
+      applicationModel.addApplication(applicationTwo),
+      applicationModel.addApplication(applicationThree),
+      applicationModel.addApplication(applicationFour),
+      applicationModel.addApplication(applicationFive),
+    ];
+    const [id1, id2, id3, id4, id5] = await Promise.all(promises);
+    applicationId1 = id1;
+    applicationId2 = id2;
+    applicationId3 = id3;
+    applicationId4 = id4;
+    applicationId5 = id5;
+  });
+
+  after(async function () {
+    await cleanDb();
+  });
+
+  describe("GET /applications", function () {
+    it("should return all the application if the user is super user and there is no user id, and next url if the size provided is equal to the applications returned in query", function (done) {
+      chai
+        .request(app)
+        .get("/applications?size=5")
+        .set("cookie", `${cookieName}=${superUserJwt}`)
+        .end((err, res) => {
+          if (err) {
+            return done(err);
+          }
+
+          expect(res).to.have.status(200);
+          expect(res.body).to.be.a("object");
+          expect(res.body.message).to.equal("Applications returned successfully");
+          expect(res.body.applications).to.be.a("array");
```

Comment:
can we also assert that its an non empty array

Reply:
there is one line below that which makes sure, that is is not empty so this is needed